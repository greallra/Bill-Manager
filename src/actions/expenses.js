import uuid from 'uuid';
import database from '../firebase/firebase';
// import expenses2 from '../tests/fixtures/expenses';
// ADD_EXPENSE
export const addExpense = (expense) => ({
  type: 'ADD_EXPENSE',
  expense
});

//middleware thunk allows us to return a function
/*
-query db
-dispatch action

*/
export const startAddExpense = (expenseData = {}) => {
  return (dispatch) => {
    const {
      description = '',
      note = '',
      amount = 0,
      createdAt = 0
    } = expenseData;
    const expense = { description, note, amount, createdAt };

    return database.ref('expenses').push(expense).then((ref) => {
      dispatch(addExpense({
        id: ref.key,
        ...expense
      }));
    });
  };
};

// REMOVE_EXPENSE
export const removeExpense = (id) => ({
  type: 'REMOVE_EXPENSE',
  id
});

export const startRemoveExpense = ({id} = {})=>{
  return (dispatch)=>{
    return database.ref(`/expenses/${id}`).remove()
    .then((r)=>{
      console.log("yes",r);
      dispatch(removeExpense(id));
    })   
    .catch((e)=>{
      console.log("no",e);
    })       
  }
}

// EDIT_EXPENSE
export const editExpense = (id, updates) => ({
  type: 'EDIT_EXPENSE',
  id,
  updates
});
// SET_EXPENSES
export const setExpenses =(expenses = [])=>{
  return {
    type: 'SET_EXPENSES',
    expenses
  }
};

export const startSetExpenses =()=>{
  return (dispatch)=>{
    
    return database.ref('expenses').once('value')
    .then(function(dataSnapshot) {
      // handle read data.
        let expenses2 = []
        dataSnapshot.forEach(function (childSnapshot) {
          var value = childSnapshot.val();
          expenses2.push({
            id: childSnapshot.key,
            ...value
          })
        });
        dispatch(setExpenses(expenses2));
    })
  }
};



