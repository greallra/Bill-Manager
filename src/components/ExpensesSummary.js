
import React from 'react';
import getExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses'
import { connect } from 'react-redux';
import numeral from 'numeral';
// total here: {getExpensesTotal()}

export class ExpensesSummary extends React.Component {
    render() {
        const expenseWord = this.props.expenseCount ===1 ? 'expense':'expenses';
        const formattedTotal = numeral(this.props.expensesTotal / 100).format('$0,0.00')
        return(
            <div>
                <h3>Total: {formattedTotal} {expenseWord}</h3>

            </div>
        )
    }
}
//expenses: selectExpenses(state.expenses, state.filters)
const mapStateToProps = (state)=>{
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expensesTotal: getExpensesTotal(visibleExpenses),
        expenseCount: visibleExpenses.length
    }
}

export default connect(mapStateToProps)(ExpensesSummary);