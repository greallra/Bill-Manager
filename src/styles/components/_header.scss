header {
    background-color: #2b2222;
}

.banner {
    display: flex;
    justify-content: space-between;
    color: white;
    max-width: 800px;
    margin: auto;
    button {
        background-color: #2b2222;
        color: white;
        border: none;
    }
}