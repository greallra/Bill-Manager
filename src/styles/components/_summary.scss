.summary-cont {
    padding: 20px 100px;
    background-color: #f7f7f7;
}
h3 {
    margin: 0;
}
