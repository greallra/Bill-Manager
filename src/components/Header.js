import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

const Header = ({startLogout}) => (
  <header>
    <div className="banner">
      <h1>Bill Manager</h1>
      <button onClick={startLogout}>Logout</button>
    </div>
    {/* <NavLink to="/dashboard" activeClassName="is-active">Dashboard</NavLink>
    <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
    <NavLink to="/help" activeClassName="is-active">Help</NavLink> */}
    
  </header>
);

const mapDispatchToProps =(dispatch)=>(
  {
    startLogout:()=>dispatch(startLogout())
  }
)
export default connect(undefined, mapDispatchToProps)(Header);
