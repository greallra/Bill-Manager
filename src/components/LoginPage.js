import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth'

export function LoginPage(props) {
    return <div className="box_layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Bill Manger</h1>
            <p>Get your bills under control</p>
            <button onClick={props.startLogin}>
                login

            </button>
        </div>
    </div>
}

const mapDispatchToProps =(dispatch)=>{
    return {
        startLogin: ()=>dispatch(startLogin())
    }
}

export default connect(undefined, mapDispatchToProps)(LoginPage);