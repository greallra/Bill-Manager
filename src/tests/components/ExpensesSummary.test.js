import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>);
  expect(wrapper).toMatchSnapshot();
});
test('should render ExpensesSummary correctly many expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={235555}/>);
  expect(wrapper).toMatchSnapshot();
});
