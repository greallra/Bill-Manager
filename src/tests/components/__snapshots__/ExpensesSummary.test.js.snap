// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly many expenses 1`] = `
<div>
  <h3>
    Total: 
    $2,355.55
     
    expenses
  </h3>
</div>
`;

exports[`should render ExpensesSummary correctly with 1 expense 1`] = `
<div>
  <h3>
    Total: 
    $2.35
     
    expense
  </h3>
</div>
`;
